### PR TITLE
set application name in call screens from String nc_app_name

### PR DESCRIPTION
**Pull request by mahibi**
Signed-off-by: Marcel Hibbe <dev@mhibbe.de>

### CODE REVIEW

**Comment by mahibi** [comment]
by the way i realized https://github.com/nextcloud/talk-android/issues/1064 but this should be done later

File: app/src/main/java/com/nextcloud/talk/controllers/CallController.java [hunk 1]
```diff
@@ -2360,6 +2360,15 @@ private void setCallState(CallStatus callState) {
         }
     }
 
+    private String getDescriptionForCallType() {
+        String appName = Objects.requireNonNull(getResources()).getString(R.string.nc_app_name);
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/CallController.java:2364]: If you think this might be null, then you also need to handle resulting NPE.

File: app/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -166,9 +166,9 @@
     <string name="nc_permissions_settings">Einstellungen öffnen</string>
 
     <!-- Call -->
-    <string name="nc_voice_call">Nextcloud Talk Sprachanruf</string>
```
Comment (tobiasKaminsky) [inline app/src/main/res/values-de/strings.xml:169]: This will be overwritten again via Transifex, so please do not commit this.
Best practice is not to change an already existing string, but use a new one.
Otherwise you might have strange UX if a language is not updated.
With having a new string, worst what could happen is that it is English…

File: app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java [hunk 3]
```diff
@@ -343,11 +343,15 @@ private void runAllThings() {
         showAnswerControls();
     }
 
-    @SuppressLint("LongLogTag")
+    @SuppressLint({"LongLogTag", "SetTextI18n"})
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java:346]: Why is this added?

File: app/src/main/java/com/nextcloud/talk/controllers/CallController.java (hunk 1 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/CallController.java]: ah forgot to change this. actually we don't check anywhere for NPE for resources. so i stick with this. or any concerns?

File: app/src/main/res/values-de/strings.xml (hunk 2 above)
Comment (mahibi) [reply app/src/main/res/values-de/strings.xml]: ah sure. renamed the keys and changed the values

File: app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java (hunk 3 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java]: not needed anymore

File: app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java [hunk 4]
```diff
@@ -343,11 +343,15 @@ private void runAllThings() {
...
 
-    @SuppressLint("LongLogTag")
+    @SuppressLint({"LongLogTag"})
     @Override
     protected void onViewBound(@NonNull View view) {
         super.onViewBound(view);
 
+        String callDescriptionWithoutTypeInfo =
+                getResources().getString(R.string.nc_app_name) + " " + getResources().getString(R.string.nc_call_unknown);
+        incomingCallVoiceOrVideoTextView.setText(callDescriptionWithoutTypeInfo);
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java:353]: String concatenation are normally done as a separate string with "%s1 and %s2" so that left to right languages work.

Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java:353]: but here i combine two translated strings. it would be possible to make another string that has the value "%1$s %2$s" so for other languages it could be changed to "%2$s %1$s", but i don't think anyone will understand that?
or do i get it wrong?

Comment (tobiasKaminsky) [reply app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java:353]: True, this is hard to understand on translator side.
Maybe then have "%s call", where %s is app name?

Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/CallNotificationController.java:357]: ah sure 🤦‍♂️ 😆
